rust-runtime/aws-smithy-runtime/src/client/orchestrator/auth.rs

```diff
@@ -190,6 +183,86 @@ pub(super) async fn orchestrate_auth(
     Err(NoMatchingAuthSchemeError(explored).into())
 }
 
+pub(super) fn sign_request(
+    scheme_id: AuthSchemeId,
+    identity: &Identity,
+    ctx: &mut InterceptorContext,
+    runtime_components: &RuntimeComponents,
+    cfg: &ConfigBag,
+) -> Result<(), BoxError> {
+    trace!("signing request");
+    let request = ctx.request_mut().expect("set during serialization");
+    let endpoint = cfg
+        .load::<Endpoint>()
+        .expect("endpoint added to config bag by endpoint orchestrator");
+    let auth_scheme = runtime_components
+        .auth_scheme(scheme_id)
+        .ok_or("should be configured")?;
+    let signer = auth_scheme.signer();
+    let auth_scheme_endpoint_config = extract_endpoint_auth_scheme_config(endpoint, scheme_id)?;
+    trace!(
+        signer = ?signer,
+        "signing implementation"
+    );
+    signer.sign_http_request(
+        request,
+        identity,
+        auth_scheme_endpoint_config,
+        runtime_components,
+        cfg,
+    )?;
+    Ok(())
+}
+
+// Marker indicating that endpoint resolution for determining the auth scheme option
+// has already been performed by a `AuthSchemeOptionResolver`, such as `EndpointBasedAuthSchemeOptionResolver`.
+//
+// When this marker is present in the config bag (the default for SDKs moving forward),
+// `resolve_identity` will skip endpoint resolution.
+#[doc(hidden)]
+#[derive(Clone, Debug)]
+pub struct AuthSchemeOptionResolverHandlesEndpointResolution;
+
+impl Storable for AuthSchemeOptionResolverHandlesEndpointResolution {
+    type Storer = StoreReplace<Self>;
+}
+
+// Checks whether an `endpoint` resolved by `SharedEndpointResolver` in `runtime_components`,
+// using `EndpointResolverParams` from `cfg`, contains `scheme_id` in its `authSchemes` property.
+//
+// Returns `Ok` if the condition is met, with the `Ok` variant also containing the actual `endpoint` used for verification.
```

Comment:
This returns `Ok` without the endpoint as well...description/intent here needs honed and cleaned up.

Reply:
Updated comment in 40b5e26